arviz/plots/backends/matplotlib/distplot.py

```diff
@@ -87,12 +88,22 @@ def plot_dist(
 
 def _histplot_mpl_op(values, values2, rotated, ax, hist_kwargs):
     """Add a histogram for the data to the axes."""
+    if isinstance(values, xr.Dataset):
+        raise ValueError(
+            "Cannot directly convert xarray.Dataset to numpy array."
+            " Instead, create an xarray.DataArray first "
+            "or Use plot_posterior, plot_density, plot_joint"
+            "or plot_pair instead of plot_dist"
+        )
+    if isinstance(values, InferenceData):
+        raise ValueError(" Inference Data object detected. Use plot_posterior instead of plot_dist")
+
     if values2 is not None:
         raise NotImplementedError("Insert hexbin plot here")
 
     bins = hist_kwargs.pop("bins")
```

Comment:
I have seen that [bokeh file](https://github.com/arviz-devs/arviz/blob/master/arviz/plots/backends/bokeh/distplot.py#L115) does:

```
if bins is None:
        bins = get_bins(values)
```

It would be great to add this here so both backends use the same bins as default and plot looks as similar as possible independently of backend.

Reply:
Yeah sure, I will do the needful.